internal/models/user.go

```diff
@@ -573,10 +573,9 @@ func IsDuplicatedEmail(tx *storage.Connection, email, aud string, currentUser *U
 
 	for _, userID := range userIDs {
 		user, err := FindUserByID(tx, userID)
-		if err != nil && !IsNotFoundError(err) {
+		if err != nil {
 			return nil, errors.Wrap(err, "unable to find user from email identity for duplicates")
 		}
-
 		if user.Aud == aud {
```

Comment:
Why this change? Shouldn't it be more like:

```
if err != nil && !IsNotFoundError(err) {
// ...
}

if user != nil && user.Aud == aud {
// ...
```

Reply:
that would swallow the error though - we want it to return an error if there's an identity but no user found

Reply:
it shouldn't be possible for an identity to exist without a user

Reply:
I think the current state is fine. If we want to consider the approach of `user != nil && user.Aud == aud` I guess another option could be:

```
if err != nil && !IsNotFoundError(err) {
// ...
// same as err != nil && IsNotFoundError(err)
// which should have the same effect as current commit except more verbose
} else if IsNotFoundError(err) [
// ...
}

if user != nil && user.Aud == aud {
// ...
```